src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicyExtensions.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Cors.Infrastructure
+{
+    internal static class CorsPolicyExtensions
+    {
+        private static string _WildcardSubdomain = "*.";
+
+        public static bool IsOriginAnAllowedSubdomain(this CorsPolicy policy, string origin)
+        {
+            if (policy.Origins.Contains(origin))
+            {
+                return true;
+            }
+            var originUri = new Uri(origin, UriKind.Absolute);
+            return policy.Origins
+                .Where(o => o.Contains(_WildcardSubdomain))
```

Comment:
@Eilon, according to the [docs](https://msdn.microsoft.com/en-us/library/dy85x1sa(v=vs.110).aspx), `string.Contains` does not take an `StringComparison` argument. But it does perform an ordinal case-sensitive comparison anyway when searching for the substring. Are you happy with this?